src/pages/TCY/queries/useTcyClaims.tsx

```diff
@@ -0,0 +1,85 @@
+import { fromAccountId, fromAssetId } from '@shapeshiftoss/caip'
+import { poolAssetIdToAssetId } from '@shapeshiftoss/swapper'
+import { isUtxoChainId } from '@shapeshiftoss/utils'
+import { useQueries } from '@tanstack/react-query'
+import axios from 'axios'
+import { useMemo } from 'react'
+
+import type { Claim, TcyClaimer } from '../components/Claim/types'
+
+import { getConfig } from '@/config'
+import { getChainAdapterManager } from '@/context/PluginProvider/chainAdapterSingleton'
+import { isSome } from '@/lib/utils'
+import {
+  getThorchainSaversPosition,
+  isSupportedThorchainSaversAssetId,
+} from '@/state/slices/opportunitiesSlice/resolvers/thorchainsavers/utils'
+import { selectAccountIdsByAccountNumberAndChainId } from '@/state/slices/selectors'
+import { useAppSelector } from '@/state/store'
+
+export const useTCYClaims = (accountNumber: number) => {
+  const accountIdsByAccountNumberAndChainId = useAppSelector(
+    selectAccountIdsByAccountNumberAndChainId,
+  )
+
+  const accountIds = useMemo(
+    () =>
+      Object.values(accountIdsByAccountNumberAndChainId[accountNumber] || {})
+        .flat()
+        .filter(isSome),
+    [accountNumber, accountIdsByAccountNumberAndChainId],
+  )
+
+  return useQueries({
+    queries: accountIds.map(accountId => ({
+      queryKey: ['tcy-claims', accountId],
+      queryFn: async (): Promise<Claim[]> => {
+        const activeAddress = await (async () => {
+          // UTXO-based chains are the odd ones, for all address-based, we can simply use the `account` caip-10 part
+          if (!isUtxoChainId(fromAccountId(accountId).chainId))
+            return fromAccountId(accountId).account
+
+          const chainId = fromAccountId(accountId).chainId
+          const assetId = getChainAdapterManager().get(chainId)?.getFeeAssetId()
+
+          if (!assetId) return
+          if (!isSupportedThorchainSaversAssetId(assetId)) return
+
+          // TODO(gomes): this may or may not become an issue depending on how TCY aggregates things across THORFi
+          // We may have to introspect all THORFi bits and get positions across all of them, which *may* have different addresses for UTXOs (account-based)
+          // That should be super edge-case, as most users should have their position account_index 0, but there may be a selected few unlucky which do not
+          // (got into savers before we defaulted to 0th account_index addy)
+          // Wait and see when this goes live
```

Comment:
Think there's two ways we can go here:

1. Use `getThorchainFromAddress()` which *does* default to account 0, however since the whole intent here is to handle the fact that users may have savers, lending and LP positions over diff `account_index` indexes, that wouldn't work, we'd have to duplicate this over the three 
2. Assume `account_index` 0 and get that address

Tbh, the whole thing is ultra paranoia, I don't have the historical context but I'm not even sure users are active on THORFi on a non-0 `account_index` for account 0. Highly recommend we go with the second for the sake of simplicity, WDWT?

Reply:
Keeping this one then! Since this will default to 0, this will be the exact same as using 0th `account_index` address directly, with the added benefit of handling non-0 `account_index` savers.
Savers would be the only one affected by this anyway so disregard the `wouldn't work` above, LP and lending had been added long after we standardized the `account_index` to be 0 for all THORFi bits.